src/le.py

```
@@ -1178,6 +1180,131 @@ def scaletokb(value):
+        about CPU, and disk transfer as separated into reads & writes.
+        (vs. iostat, which shows CPU less granularly; it shows more
+         detail about per-disk IO, but does not split IO into reads and
+         writes)
+
+        Frustratingly, the level of per-disk statistics from top is
+        incredibly un-granular
+
+        We'll get physical memory details from here too
+        """
+        cpure = re.compile(r'CPU states:\s+([\d.]+)\% idle,\s+([\d.]+)\% user,\s+'
+                           r'([\d.]+)\% kernel,\s+([\d.]+)\% iowait')
+        memre = re.compile(r'Memory:\s+(\d+\w+) phys mem, (\d+\w+) free mem, '
+                           r'(\d+\w+) total swap, (\d+\w+) free swap')
+
+        # scaling routine for use in map() later
+        def scaletokb(value):
+            # take a value like 1209M or 10G and return an integer
+            # representing the value in kilobytes
+
+            (size, scale) = re.split('([A-z]+)', value)[:2]
+            size = int(size)
+            if scale:
+                if scale in self.scale2kb:
+                    size *= self.scale2kb[scale]
+                else:
+                    log.warning("Error: value in %s expressed in "
+                                "dimension I can't translate to kb: %s %s",
+                                line, size, scale)
+            return size
+
+        # the first set of 'top' headers display average values over
+        # system uptime.  so we only want to read the second set that we
+        # see.
+        toppass = 0
+
+        # we should really do this first, so that we don't waste any time
+        # if top fails to work.  however, it 'reads' better at this point
+        try:
+            proc = subprocess.Popen(['top',
+                                     '-i', '2', '-l', '2', '-n', '0'],
+                                    stdout=subprocess.PIPE)
+        except:
+            return
```

Should still fail gracefully if `top` is not available.